crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -110,20 +114,37 @@ impl ParsedQProperty {
                 let mut write = None;
                 let mut notify = None;
 
+                let mut constant = false;
+                let mut required = false;
+                let mut is_final = false;
+                let mut reset = None;
+
+                let map_auto_or_custom = |variable: &mut Option<FlagState>, value: &Option<Ident>| {
+                    *variable = Some(value.as_ref().map_or(FlagState::Auto, |ident| FlagState::Custom(ident.clone())));
+                };
+
                 // Create mutable closure to capture the variables for setting with the Meta values
                 let mut update_fields = |ident: &Ident, value: Option<Ident>| -> Result<()> {
-                    let variable = match ident.to_string().as_str() {
-                        "read" => &mut read,
-                        "write" => &mut write,
-                        "notify" => &mut notify,
+                    match ident.to_string().as_str() {
+                        "READ" => map_auto_or_custom(&mut read, &value),
+                        "WRITE" => map_auto_or_custom(&mut write, &value),
+                        "NOTIFY" => map_auto_or_custom(&mut notify, &value),
+                        "CONSTANT" => constant = true,
+                        "REQUIRED" => required = true,
+                        "FINAL" => is_final = true,
+                        "RESET" => reset = Some(
+                                value.ok_or(Error::new(
+                                    ident.span(),
+                                    "Invalid flag passed, must be one of READ, WRITE, NOTIFY, RESET, CONSTANT, REQUIRED, FINAL",
+                                ))?
+                            ),
                         _ => {
                             return Err(Error::new(
                                 ident.span(),
-                                "Invalid flag passed, must be one of read, write, notify",
+                                "Invalid flag passed, must be one of READ, WRITE, NOTIFY, RESET, CONSTANT, REQUIRED, FINAL",
                             ));
```

Comment:
there is the same error here and instead of `ok_or` we should try to use `ok_or_else` so that it doesn't construct the error when not required.

Instead can we have a closure for the error?

```rust
let unknown_flag_error = || {
   Error::new(
        ident.span(),
       "Invalid flag passed, must be one of READ, WRITE, NOTIFY, RESET, CONSTANT, REQUIRED, FINAL",
  )
};

...

reset = Some(value.ok_or_else(unknown_flag_error));

...

_ => return Err(unknown_flag_error())
```

Reply:
done

Reply:
however i guess for the RESET one you could instead be saying that RESET requires a value and cannot have RESET rather than `RESET = myFunction` as that is what the actual error is in that case?

So in that case the closure would go and you would return like before, but use `ok_or_else(|| Error...)`, and update the message, what do you think?